test/distributed/test_jit_c10d.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+
+import unittest
+import traceback
+from sys import platform
+import torch
+import torch.distributed as c10d
+
+from torch.testing._internal.common_distributed import MultiProcessTestCase, \
+    requires_gloo
+from torch.testing._internal.common_utils import TEST_WITH_TSAN
+
+from torch.testing._internal.jit_utils import JitTestCase, _inline_everything
+
+
+if not c10d.is_available():
+    print('c10d not available, skipping tests', file=sys.stderr)
+    sys.exit(0)
+
+
+if platform == 'darwin':
+    LOOPBACK = 'lo0'
+else:
+    LOOPBACK = 'lo'
+
+@requires_gloo()
+@unittest.skipIf(TEST_WITH_TSAN, "TSAN is not fork-safe since we're forking in a multi-threaded environment")
+class ProcessGroupGlooJitTest(MultiProcessTestCase):
+    def setUp(self):
+        super(ProcessGroupGlooJitTest, self).setUp()
+        self._fork_processes()
+
+    def opts(self, threads=2):
+        opts = c10d.ProcessGroupGloo.Options()
+        opts.devices = [c10d.ProcessGroupGloo.create_device(interface=LOOPBACK)]
+        opts.timeout = 5.0
+        opts.threads = threads
+        return opts
+
+    def test_jit_process_group(self):
+        @torch.jit.script
+        def test_create_store(filename: str, world_size: int) -> torch.classes.dist_c10d.FileStore:
+            store = torch.classes.dist_c10d.FileStore(filename, world_size)
```

Comment:
Is this what the user facing API would look like with torchbind? Is there a way to get around this and use `torch.distributed.FileStore`? The reason being that this is kind of an API change for the user if they would like to use the same code under TorchScript.

Reply:
ultimately we would like to make `torch.distributed.FileStore = torch.classes.dist_c10d.FileStore`, this way the user could just reuse the existing API either in python and in TorchScript. But this currently is not possible due to various TorchBind limitations (e.g. [support `std::chrono:milisecond` as argument](https://github.com/pytorch/pytorch/issues/42948), [install doc via TorchBind](https://github.com/pytorch/pytorch/issues/42951), etc.) So I am just purely testing it here to make sure the TorchBind part works fine, and we will improve the TorchBind to cover those missing features before fully move to TorchBind